libparsec/crates/platform_storage/src/web/certificates.rs

```diff
@@ -62,36 +303,69 @@ impl<'a> PlatformCertificatesStorageForUpdateGuard<'a> {
 }
 
 #[derive(Debug)]
-pub(crate) struct PlatformCertificatesStorage {}
+pub(crate) struct PlatformCertificatesStorage {
+    conn: Arc<IdbDatabase>,
+}
+
+// Safety: PlatformCertificatesStorage is read only
+unsafe impl Send for PlatformCertificatesStorage {}
 
 impl PlatformCertificatesStorage {
     pub async fn no_populate_start(
         data_base_dir: &Path,
         device: &LocalDevice,
     ) -> anyhow::Result<Self> {
-        todo!()
+        // 1) Open the database
+
+        #[cfg(feature = "test-with-testbed")]
+        let name = format!(
+            "{}-{}-certificates",
+            data_base_dir.to_str().unwrap(),
+            device.slug()
+        );
+
+        #[cfg(not(feature = "test-with-testbed"))]
+        let name = format!("{}-certificates", device.slug());
+
+        let db_req =
+            IdbDatabase::open_u32(&name, DB_VERSION).map_err(|e| anyhow::anyhow!("{e:?}"))?;
+
+        // 2) Initialize the database (if needed)
+
+        let conn = Arc::new(super::model::initialize_model_if_needed(db_req).await?);
+
+        // 3) All done !
+
+        let storage = Self { conn };
+        Ok(storage)
     }
 
     pub async fn stop(self) -> anyhow::Result<()> {
-        todo!()
+        Ok(())
```

Comment:
Why we don't call [`idb.close()`](https://developer.mozilla.org/fr/docs/Web/API/IDBDatabase/close) here ?

Reply:
I didn't even know check, they don't use it on their example :grimacing:

Reply:
Done